### PR TITLE
Mtdsa 4470: Models and data types for "Submit Self Employment BSAS" endpoint.

### PR DESCRIPTION
**Pull request by BetaDraconis**

### CODE REVIEW

File: app/v1/models/request/submitBsas/selfEmployment/SeAdditions.scala [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package v1.models.request.submitBsas.selfEmployment
+
+import play.api.libs.json.{JsPath, Json, OWrites, Reads}
+import play.api.libs.functional.syntax._
+
+case class SeAdditions(costOfGoodsBoughtDisallowable: Option[BigDecimal],
```
Comment (narendravijayarao) [inline app/v1/models/request/submitBsas/selfEmployment/SeAdditions.scala]: Do we need **_Se_** for the file names? they already separated via package

Comment (BetaDraconis) [reply app/v1/models/request/submitBsas/selfEmployment/SeAdditions.scala]: Income and Expenses have the "se" prefix to align with FHLExpenses, and FHLIncome in the other endpoint, and I decided to put it on Additions as well just for consistency. They don't necessarily need it, but it does make it clearer which model you're importing and stops Inteliji from getting confused and importing the wrong model.

File: app/v1/models/response/SubmitSeBsasResponse.scala [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+import v1.hateoas.{HateoasLinks, HateoasLinksFactory}
+import v1.models.hateoas.{HateoasData, Link}
+
+case class SubmitSeBsasResponse(id: String)
+
+object SubmitSeBsasResponse extends HateoasLinks {
+
+  implicit val writes: OWrites[SubmitSeBsasResponse] = Json.writes[SubmitSeBsasResponse]
+
+  implicit val reads: Reads[SubmitSeBsasResponse] =
```
Comment (narendravijayarao) [inline app/v1/models/response/SubmitSeBsasResponse.scala]: To handle **_### RULE_UK_PROPERTY_ADJUSTED_** we need **_typeOfBusiness_**
can you do something like this ---- https://github.com/hmrc/self-assessment-bsas-api/blob/264b1a05d9c421ed4a576993eb2e51819b43bc37/app/v1/models/response/SubmitUkPropertyBsasResponse.scala#L40

Comment (narendravijayarao) [reply app/v1/models/response/SubmitSeBsasResponse.scala]: https://github.com/hmrc/self-assessment-bsas-api/blob/264b1a05d9c421ed4a576993eb2e51819b43bc37/app/v1/models/response/SubmitUkPropertyBsasResponse.scala#L30

Comment (BetaDraconis) [reply app/v1/models/response/SubmitSeBsasResponse.scala]: Will do

**Pull request merged**